tests/async_tests/test_interfaces.py

```diff
@@ -210,9 +204,14 @@ async def test_http_proxy(proxy_server: URL, proxy_mode: str) -> None:
         assert reason == b"OK"
 
 
-@pytest.mark.asyncio
-# This doesn't run with trio, since trio doesn't support local_address.
 async def test_http_request_local_address() -> None:
+    exc = getattr(sniffio, "Async" + "LibraryNotFoundError")
+    try:
+        if sniffio.current_async_library() == "trio":
+            pytest.skip("The trio backend does not support local_address")
+    except exc:
+        pass
```

Comment:
One way to make this work would be to use the same trick than `lookup_async_backend()`/`lookup_sync_backend()` that we already have, e.g. have these helpers:

```python
# tests/utils.py
def lookup_async_library():
    return sniffio.current_async_library()

def lookup_sync_library():
    return "sync"
```

Then in async test cases:

```suggestion
    if lookup_async_library() == "trio":
        pytest.skip("The trio backend does not support local_address")
```

Add `("lookup_async_library", "lookup_sync_library")` to `unasync.py`, and things _should_ fall into place nicely?

Reply:
I added the module you suggested and changed the test accordingly.